Modules/_interpchannelsmodule.c

```diff
@@ -2340,6 +2340,47 @@ channel_id_converter(PyObject *arg, void *ptr)
     return 1;
 }
 
+/*[clinic input]
+module _interpchannels
+[clinic start generated code]*/
+/*[clinic end generated code: output=da39a3ee5e6b4b0d input=10a1a1ec55663e0b]*/
+
+
+/*[python input]
+
+class channel_id_arg_converter(CConverter):
+    type = 'int64_t'
+    converter = 'channel_id_arg_converter'
+    broken_limited_capi = True
+
+    def parse_arg(self, argname, displayname, *, limited_capi):
+        assert not limited_capi
+        return self.format_code("""
+            if (!{converter}(module, {argname}, &{paramname})) {{{{
+                goto exit;
+            }}}}
+            """,
+            argname=argname,
+            converter=self.converter)
+
+
+[python start generated code]*/
+/*[python end generated code: output=da39a3ee5e6b4b0d input=78e17eef989afa6b]*/
+
+static int
+channel_id_arg_converter(PyObject *module, PyObject *arg, void *ptr)
```

Comment:
You don't need to use a void pointer here. (Implying you can also get rid of the casting and temporary `cid_ptr` stuff below.

```suggestion
channel_id_arg_converter(PyObject *module, PyObject *arg, int64_t *cid)
```